libtransmission/rpc-server.c

```diff
@@ -1138,7 +1139,14 @@ tr_rpc_server* tr_rpcInit(tr_session* session, tr_variant* settings)
     }
     else
     {
-        s->url = tr_strdup(str);
+        if (str[strlen(str) - 1] != '/')
```

Comment:
Minor but you could get the `str` length from the `tr_variantDictFindStr` call above instead of calculating it yourself.

Less minor, be prepared for `str` to be empty, in which case you're currently accessing the memory outside of the string.

Reply:
Fixed. As I see from variant.c code, if we successfully found an item in a dictionary and if it is string, the pointer cannot be null even if the string is empty. Please tell me if I'm wrong.

Reply:
JSON "null" is not a string, so that's indeed impossible. If there is a string, it's not null but could still be empty which my comment was about.